tests/test_input_output.py

```diff
@@ -110,6 +110,132 @@ def test_write_desc_input_Nones(tmpdir_factory):
     assert no_gtol  # fail test if gtol line is written
 
 
+@pytest.mark.unit
+def test_descout_to_input(tmpdir_factory):
+    """
+    Test converting DESC output to a DESC input file.
+
+    To do that, we convert a DESC output file to a DESC input file
+    named desc_input_converted. We compare the boundary Fourier
+    coefficients of this file with the true values of the Fourier
+    coefficients given in desc_input_truth.
+    """
+    outfile_path = "./tests/inputs/LandremanPaul2022_QA_reactorScale_lowRes.h5"
+    tmpdir = tmpdir_factory.mktemp("desc_inputs")
+    tmp_path = tmpdir.join("input_LandremanPaul_QA")
+    tmpout_path = tmpdir.join("LandremanPaul2022_QA_reactorScale_lowRes.h5")
+    shutil.copyfile(outfile_path, tmpout_path)
+
+    ir = InputReader()
+    ir.descout_to_input(str(tmp_path), str(tmpout_path))
+    desc_input_truth = ".//tests//inputs//LandremanPaul2022_QA_reactorScale_lowRes"
+    desc_input_converted = tmp_path
+
+    with open(desc_input_truth) as f:
+        lines_correct = f.readlines()
+    with open(desc_input_converted) as f:
+        lines_converted = f.readlines()
+
+    # Count the lines before the boundary modes
+    counter0 = int(0)
+    found0 = int(0)
+    while found0 != 1:
+        if (
+            lines_correct[counter0].split("\n")[0].replace(" ", "")
+            == "#fixed-boundarysurfaceshape"
+        ):
+            found0 = 1
+            counter0 += 1
+        else:
+            counter0 += 1
+
+    counter1 = int(0)
+    found0 = int(0)
+    while found0 != 1:
+        if (
+            lines_converted[counter1].split("\n")[0].replace(" ", "")
+            == "#fixed-boundarysurfaceshape"
+        ):
+            found0 = 1
+            counter1 += 1
+        else:
+            counter1 += 1
+
+    # Tells us if the m mode number comes before n in the input file.
+    mfirst0 = (
+        lines_correct[counter0].split("\t")[1].split(":")[0].replace(" ", "") == "m"
+    )
+
+    found0 = 1
+
+    i = int(counter1)
+    j = int(counter0)
+
+    # Compare the Fourier modes and ensure that they match, i.e,, found0 = 1
+    if mfirst0 is True:
```

Comment:
the more "pythonic" way would just be `if mfirst0:`, which also guards against the possibiltiy that `mfirst0` may not be the python literal `True` (remember, `is` checks if they're the same instance), but possibly some other boolean type like a np.array of bool etc.

Reply:
Done